Heroic-Games-Launcher/HeroicGamesLauncher
# [Refactor] Sidebar cleanup (and a bit more that's related)

This PR is a refactor to cleanup the code related to the Sidebar component.

I does a few things:
- Extracted a new `SidebarItem` component, so we don't need to use `<button>` and `NavLink` directly in SidebarLinks (this helps remove a lot of duplication around creating those items)
- Replaced the .scss files with .css, we are not really using any sass feature
- Refactored the css files to use native css nesting and moved the styles to the corresponding components' css (we had a lot of css in the main `sidebar.css` file that is easier to find in each component's css, we also had css for `Sidebar__item` in multiple css files)
- I added new CSS properties specific to the sidebar that can help when theming, they fallback to what we already had so nothing changes, but new themes can use better variables that don't cause unexpected side effect
- Because of this cleanup, I had to remove the `state: { fromGameCard: ...}` that was not really used anymore since the settings for a game were moved to the SettingsModal component. This led into some changes needed also in the `useSettingsContext` context and the `Settings` screen to remove the code that was using that

**How to test/qa this**

The sidebar should behave the same way as before (I didn't really change any CSS, I just moved things to different places and added nesting, and the new custom properties have fallbacks to the previous values):
- Resizing
- Hiding content during resize
- sizes, colors, etc depending on the theme
- etc

---

Use the following Checklist if you have changed something on the Backend or Frontend:
